### ISSUE TITLE
@angular-eslint/schematics 12.2.0 - linting warnings when using typescript 4.3.2

### ISSUE THREAD

**Issue by gauravshah27** [angular-eslint/angular-eslint#557]
Versions:

I am building an @angular/cli@12.1.0 based project and that uses typescript 4.3.2 by default. 

when I run the command 
`ng add @angular-eslint/schematics`

Following version numbers get added to my package.json:

```javascript
  "@angular-eslint/builder": "12.2.0",
  "@angular-eslint/eslint-plugin": "12.2.0",
  "@angular-eslint/eslint-plugin-template": "12.2.0",
  "@angular-eslint/schematics": "12.2.0",
  "@angular-eslint/template-parser": "12.2.0",
  "@typescript-eslint/eslint-plugin": "4.23.0",
  "@typescript-eslint/parser": "4.23.0",
```

When I run the command 
`npm run lint`  

I get the following warning in my console:

```dos
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.3.0

YOUR TYPESCRIPT VERSION: 4.3.4

Please only submit bug reports when using the officially supported version.

=============

All files pass linting.
```

I had to manually update the versions of `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` to the latest version - 4.28.0 for it to work correctly and not give me that warning.

I am just wondering whether @angular-eslint/schematics will need a patch version upgrade to put the correct versions of the @typescript-eslint/* packages in the package.json when we are using the schematic to add eslint support?
